src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -358,14 +366,6 @@ private function createFirewall(ContainerBuilder $container, string $id, array $
 
         $config->replaceArgument(5, $defaultProvider);
 
-        // Register Firewall-specific event dispatcher
-        $firewallEventDispatcherId = 'security.event_dispatcher.'.$id;
-        $container->register($firewallEventDispatcherId, EventDispatcher::class);
```

Comment:
I would keep the registration of the dispatcher service here instead of moving it up (you only moved it partially anyway). Move only the `firewallEventDispatcherId` variable up.

Reply:
👍